### PR TITLE
Updating Clio libraries on account of security vulnerability

### PR DESCRIPTION
**Pull request by rhiananthony**
### Description

There are three tickets related to general security fixes GH-624, GH-625, GH-626. These required updating several libraries to newer versions, which in turn led to many other updates. 

 s
----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: Ensure that you have added or updated tests
- [ ] **Submitter**: If you're adding/switching libraries or otherwise changing the design, update the [design documentation](https://broadinstitute.atlassian.net/wiki/pages/viewpage.action?pageId=114531509).
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
* Review cycle:
  * Team may comment on PR at will
  * **Reviewer assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to reviewer** for further feedback
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Move the JIRA issue to QA once this checklist is completed

### CODE REVIEW

**Review by pshapiro4broad — APPROVED** [review]

**Review by tbl3rd — APPROVED** [review]
Yow.

File: clio-client/src/test/scala/org/broadinstitute/clio/client/webclient/ClioWebClientSpec.scala [hunk 1]
```diff
@@ -143,7 +143,7 @@ class ClioWebClientSpec extends BaseClientSpec with AsyncMockFactory {
   def jsonResponse[A: Encoder](body: A): HttpResponse = {
     val entity = HttpEntity(
       ContentTypes.`application/json`,
-      body.asJson.pretty(defaultPrinter)
+      body.asJson.printWith(defaultPrinter)
```
Comment (rexwangcc) [inline clio-client/src/test/scala/org/broadinstitute/clio/client/webclient/ClioWebClientSpec.scala:146]: It's a bit surprising that this type of interface change wasn't backward compatible.

File: clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/RecoveryIntegrationSpec.scala [hunk 2]
```diff
@@ -160,7 +160,7 @@ class RecoveryIntegrationSpec extends DockerIntegrationSpec with OptionValues {
                 )
                 .fold(throw _, identity)
 
-              s"$dateDir/${upsertId.persistenceFilename}" -> defaultPrinter.pretty(json)
+              s"$dateDir/${upsertId.persistenceFilename}" -> defaultPrinter.print(json)
```
Comment (rexwangcc) [inline clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/RecoveryIntegrationSpec.scala:163]: `print` or `printWith`?

File: clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/PersistenceDAO.scala [hunk 3]
```diff
@@ -84,7 +84,7 @@ abstract class PersistenceDAO(recoveryParallelism: Int) extends LazyLogging {
 
     val upsertId = ElasticsearchIndex.getUpsertId(document)
 
-    val jsonString = ModelAutoDerivation.defaultPrinter.pretty(document)
+    val jsonString = ModelAutoDerivation.defaultPrinter.print(document)
```
Comment (rexwangcc) [inline clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/PersistenceDAO.scala:87]: same as above, although it looks like `document` and `defaultPrinter` now have different functions to achieve `pretty()`?

File: clio-client/src/test/scala/org/broadinstitute/clio/client/webclient/ClioWebClientSpec.scala (hunk 1 above)
Comment (rhiananthony) [reply clio-client/src/test/scala/org/broadinstitute/clio/client/webclient/ClioWebClientSpec.scala:146]: Agreed

File: clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/RecoveryIntegrationSpec.scala (hunk 2 above)
Comment (rhiananthony) [reply clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/RecoveryIntegrationSpec.scala:163]: printWith is used with json objects. This is a string.

File: clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/PersistenceDAO.scala (hunk 3 above)
Comment (rhiananthony) [reply clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/PersistenceDAO.scala:87]: Yep

**Pull request merged**